### PR TITLE
new upstream v4.1.0

### PR DESCRIPTION
**Pull request by joelkraehemann**
* implemented presets dialog
* implemented zoom of PDF manual
* implemented print of PDF manual

### CODE REVIEW

File: org.nongnu.gsequencer.gsequencer.json [hunk 1]
```diff
@@ -212,15 +254,32 @@
                 }
             },
             "post-install": [
+                "mkdir -p /app/share/doc/gsequencer/pdf/",
+                "install -c -p -m 644 pdf/user-manual-a4.pdf /app/share/doc/gsequencer/pdf/user-manual-a4.pdf",
+                "install -c -p -m 644 pdf/user-manual-letter.pdf /app/share/doc/gsequencer/pdf/user-manual-letter.pdf",
```
Comment (hfiguiere) [inline org.nongnu.gsequencer.gsequencer.json:259]: if you use `install -D` you don't need the mkdir. It's usually good practice to do so

File: org.nongnu.gsequencer.gsequencer.json [hunk 2]
```diff
@@ -212,15 +254,32 @@
...
                 "sed -i -e 's~Icon=gsequencer~Icon=org.nongnu.gsequencer.gsequencer~g' /app/share/applications/gsequencer.desktop",
```
Comment (hfiguiere) [inline org.nongnu.gsequencer.gsequencer.json:260]: there is `desktop-file-edit` to do that kind of changes, less obscure than `sed`

also with `rename-icon` it's not even necessary.

File: org.nongnu.gsequencer.gsequencer.json [hunk 3]
```diff
@@ -212,15 +254,32 @@
...
                 {
                     "type": "archive",
-                    "url": "https://download-mirror.savannah.gnu.org/releases/gsequencer/4.0.x/gsequencer-4.0.13.tar.gz",
-                    "sha256": "dba48feed9e8455c5234f002cc874bfb55e51376024046ee5faa4ed528664084"
+                    "url": "https://download-mirror.savannah.gnu.org/releases/gsequencer/4.1.x/gsequencer-4.1.1.tar.gz",
+                    "sha256": "1dbda4450fb4744d7f45acd80b8f5aff53f849f196b3343d97f1cb1a65e0c991"
+                },
+                {
+                    "type": "file",
+                    "url": "http://xn--krhemann-1za.com/download/ags-user-manual-a4.pdf",
```
Comment (hfiguiere) [inline org.nongnu.gsequencer.gsequencer.json:272]: I hope you don't plan to update it without changing the name.

File: org.nongnu.gsequencer.gsequencer.json [hunk 4]
```diff
@@ -212,15 +254,32 @@
                 }
             },
             "post-install": [
+                "mkdir -p /app/share/doc/gsequencer/pdf/",
+                "install -c -p -m 644 pdf/user-manual-a4.pdf /app/share/doc/gsequencer/pdf/user-manual-a4.pdf",
+                "install -c -p -m 644 pdf/user-manual-letter.pdf /app/share/doc/gsequencer/pdf/user-manual-letter.pdf",
                 "sed -i -e 's~Icon=gsequencer~Icon=org.nongnu.gsequencer.gsequencer~g' /app/share/applications/gsequencer.desktop",
                 "install -c -p -m 644 COPYING /app/share/gsequencer/GPL-3",
```
Comment (hfiguiere) [inline org.nongnu.gsequencer.gsequencer.json:261]: I would remove the `shell` source below that does the exact same as here. (`post-install` is the righ way)
```suggestion
                "install -c -p -Dm644 COPYING -t /app/share/gsequencer/GPL-3",
```

File: org.nongnu.gsequencer.gsequencer.json [hunk 5]
```diff
@@ -196,12 +196,54 @@
...
+                    "-DENABLE_LIBOPENJPEG='openjpeg2'",
+                    "-DENABLE_DCTDECODER='libjpeg'",
+                    "-DENABLE_SPLASH=OFF",
+                    "-DENABLE_BOOST=OFF",
+                    "-DENABLE_CPP=OFF"
+            ],
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "https://poppler.freedesktop.org/poppler-22.02.0.tar.xz",
```
Comment (hfiguiere) [inline org.nongnu.gsequencer.gsequencer.json:233]: I think 22.06 is out now

File: org.nongnu.gsequencer.gsequencer.json (hunk 3 above)
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json:270]: I am unsure what do you want?

File: org.nongnu.gsequencer.gsequencer.json (hunk 5 above)
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json]: I fixed this.

File: org.nongnu.gsequencer.gsequencer.json (hunk 4 above)
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json]: Why -t?

File: org.nongnu.gsequencer.gsequencer.json (hunk 2 above)
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json]: Removed it.

File: org.nongnu.gsequencer.gsequencer.json (hunk 3 above)
Comment (hfiguiere) [reply org.nongnu.gsequencer.gsequencer.json:270]: if you regenerate the manual (like for a future release) its sha256 will change and the build will be broken. This is what happen with non-versioned resources. tl;dr new revision = new name, and there is no indication of that here.

(this apply to the other too)

File: org.nongnu.gsequencer.gsequencer.json (hunk 1 above)
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json]: Just did it so.

File: org.nongnu.gsequencer.gsequencer.json (hunk 4 above)
Comment (hfiguiere) [reply org.nongnu.gsequencer.gsequencer.json]: maybe no `-t` here if you rename the file. I might have gone too far. Still the `-D` applies to make sure the directory exists.

File: org.nongnu.gsequencer.gsequencer.json (hunk 3 above)
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json]: Just provide PDFs with version information.

Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json]: just added version information to filenames.

**Pull request merged**